lifecycle.go

```diff
@@ -52,7 +53,16 @@ func handleLifecycleTransition(db repo, log logrus.FieldLogger,
 		return nil
 	}
 
-	err = completeLifecycleAction(action, log, asSvc)
+	if transition == "terminating" && os.Getenv("DETACH_INSTANCE_ON_TERMINATION") == "true" {
```

Comment:
This is cool for starters.  I'd like to see this env var integrated with the rest of the configuration bits if possible.